stats/binary/collector.go

```diff
@@ -0,0 +1,118 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2016 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package binary
+
+import (
+	"bufio"
+	"context"
+	"encoding/gob"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+
+	"github.com/loadimpact/k6/stats"
+	"github.com/spf13/afero"
+)
+
+// Collection represents a binary collector that writes
+// samples to a gob binary file
+type Collector struct {
+	file    io.WriteCloser
+	writer  *bufio.Writer
+	encoder *gob.Encoder
+	recvr   chan []stats.Sample
+
+	closed bool
+	lock   sync.Mutex
+}
+
+func init() {
+	// register type passed as interface{}
+	gob.Register(stats.CounterSink{})
+	gob.Register(stats.GaugeSink{})
+	gob.Register(stats.TrendSink{})
+	gob.Register(stats.RateSink{})
+	gob.Register(stats.DummySink{})
+}
+
+// New returns a collector ready to start writing samples
+func New(fs afero.Fs, fname string) (*Collector, error) {
+	if fname == "" || fname == "-" {
```

Comment:
I think the empty string (`-o bin`) should error; if you want to output binary data to stdout, it should be explicit imo.